homeassistant/components/abode/sensor.py

```diff
@@ -19,18 +21,38 @@
 from . import AbodeDevice, AbodeSystem
 from .const import DOMAIN
 
-SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
-    SensorEntityDescription(
+
+@dataclass
+class AbodeSensorDescriptionMixin:
+    """Mixin for Abode sensor."""
+
+    value_fn: Callable[[AbodeSense], float]
+    native_unit_of_measurement_fn: Callable[[AbodeSense], str]
+
+
+@dataclass
+class AbodeSensorDescription(SensorEntityDescription, AbodeSensorDescriptionMixin):
+    """Class describing Abode sensor entities."""
+
+
+SENSOR_TYPES: tuple[AbodeSensorDescription, ...] = (
+    AbodeSensorDescription(
         key=CONST.TEMP_STATUS_KEY,
         device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement_fn=lambda device: device.temp_unit,
```

Comment:
When setting a device class that has a required unit of measurement the entity must use one of our common constants. I don't believe that's the case here. There should be a map that we use to look up the Home Assistant unit from the device unit.

Reply:
The constants are the same string value as our enum. Do we still want to use a transformation map for this?

Reply:
Yes, we should never rely on code that is outside our control when we set values that have specific required values in our API.

Reply:
#99323